Atlas/backend/atlas/views.py

```diff
@@ -175,9 +175,43 @@ class cultural_heritage_item_search(generics.ListAPIView):
     permission_classes = [IsAuthenticatedOrReadOnly]
     def get_queryset(self):
         query = self.kwargs.get('query')
-        return Cultural_Heritage.objects.annotate(
-           search=SearchVector('tags__name', 'title','description'),
-        ).filter(search=query)
+        #self.keywords = hidden_tag_extractor.extract_keywords(hidden_tag_extractor,query)
+        self.keywords = query.split()
+        objects = Cultural_Heritage.objects.all()
+        objects_with_score = []
+        for object in objects:
+            objects_with_score.append((self.cmp(object),object))
+        objects_with_score =sorted(objects_with_score,key=lambda x : x[0])
+        objects_with_positive_score = []
+        for i in range(0,len(objects_with_score)):
```

Comment:
A possible ```filter``` implementation is as follows:

```python
objects_with_positive_score = filter(lambda x: x > 0, objects_with_score)
```

This returns an iterator, so it is also more efficient